### DISCUSSION TITLE
How to properly configure forge fork and fork a cloned repo?

### DISCUSSION THREAD

**Discussion by ldeck** [magit/forge#351] [Support] [answered]
I've cloned a repo from GitHub and would now like to fork it in order to create a pull request.

From Magit status (on the master branch of the clone) I've selected forge-fork (`'cf`), but I'm presented with this error:
```
Wrong type argument: listp, "ldeck"
```

forge-owned-accounts has value ("ldeck"), so what am I missing?
Thanks.

**Comment by tarsius** [comment, marked as answer]
Do `M-x toggle-debug-on-error` and try again. The resulting backtrace should help narrowing it down.

**Reply by ldeck** [reply]
Thanks @tarsius . Here's the resulting stack-trace:
```
Debugger entered--Lisp error: (wrong-type-argument listp "ldeck")
  car("ldeck")
  mapcar(car ("ldeck"))
  byte-code("\301\302\303\304\10\"\"\211\305\306\307\310\5\10\"A\311\"\206\26\0\3\"D\207" [forge-owned-accounts magit-completing-read "Fork to" mapcar car read-string "Remote name: " plist-get assoc remote-name] 8)
  call-interactively(forge-fork nil nil)
  command-execute(forge-fork)
```

Versions FYI:
```
Magit 20210326.329, Git 2.31.0, Emacs 27.2, darwin
forge              20210314.2238
```

**Reply by tarsius** [reply]
> forge-owned-accounts has value ("ldeck"), so what am I missing?

That is *not* a valid value for that variable. The format is `((ACCOUNT . PLIST)...)` PLIST may be nil, so what you want is `(("ldeck"))`. People keep getting this wrong, We'll have to explicitly check the value and give a more helpful error message.

**Reply by tarsius** [reply]
Ah I see, the information in the manual is outdated and just talks about a list.

**Reply by ldeck** [reply]
> > forge-owned-accounts has value ("ldeck"), so what am I missing?
> 
> That is _not_ a valid value for that variable. The format is `((ACCOUNT . PLIST)...)` PLIST may be nil, so what you want is `(("ldeck"))`. People keep getting this wrong, We'll have to explicitly check the value and give a more helpful error message.

Ah cool. Okay so I've updated my config (as below). Thanks for your help!

```
(use-package forge
  :ensure t
  :after magit
  :config
  (setq forge-owned-accounts '())
  (let ((accounts (remove nil
                          (list
                           (getenv "FORGE_OWNED_ACCOUNTS")
                           (getenv "USER")))))
    (if accounts
        (progn
          (message "Configuring forge-owned-accounts: %s" accounts)
          (add-to-list 'forge-owned-accounts accounts)
          ))))
```

**Reply by tarsius** [reply]
I've updated the manual and hope that is enough (i.e. I haven't added an explicit assert (yet?)).
